src/mainboard/emulation/qemu-fsp/src/main.rs

```diff
@@ -110,45 +110,50 @@ pub extern "C" fn _start(_fdt_address: usize) -> ! {
     // from this API. It is recommended for the bootloader to save this HobListPtr
     // returned from this API and parse the full HOB list after the FspSiliconInit() API."
     unsafe {
```

Comment:
Which part of this is actually unsafe? It seems like it's mainly just dereferencing hob_list_ptr. It would be good to put just that part in an unsafe block and pull the rest out.

For example you can just dereference the pointer once in an unsafe block and then safely use that variable elsewhere.

Reply:
Cleaned up while usptreaming, thanks! As suggested only the deferences are marked unsafe.